src/dune_rules/dune_env.mli

```diff
@@ -37,6 +37,8 @@ module Stanza : sig
     ; js_of_ocaml : Ordered_set_lang.Unexpanded.t Js_of_ocaml.Env.t
     ; coq : Ordered_set_lang.Unexpanded.t
     ; format_config : Format_config.t option
+    ; error_on_use : (Loc.t * string) option
+    ; warn_on_load : (Loc.t * string) option
```

Comment:
I see a couple of improvements that could be made. One is small and easy, and the other is a little more work but makes the code cleaner in the end.

1. Change string to `unit Pp.t` to reflect that this is a message
2. Change the types to `User_message.t` which is the most accurate type to use for these fields.

They are mutually exclusive, so pick one according to how much time you have.

Reply:
I started with `User_message.t` but this type needs an `equal` function, but `User_message` and `Pp` do not have one at the moment. I think that it's the best but it's a been more involved. Does that sound good to add `equal` functions there?